added jellyfish count, dump, histo, and merge subcommands

williamrowell opened the pull request:
Added commands for counting of kmers, dumping kmers, producing a kmer histogram, and merging kmer databases.

johanneskoester commented on bio/jellyfish/merge/test/Snakefile:
```diff
@@ -0,0 +1,5 @@
+rule jellyfish_merge:
```
Please format all test Snakefiles with [Snakefmt](https://github.com/snakemake/snakefmt).

williamrowell replied:
reformatted and pushed

johanneskoester reviewed (changes requested):
Thanks a lot. Just the formatting, see below.

johanneskoester reviewed (dismissed):
Sorry for the delay, I just discovered that I missed one other thing: you also need to add test cases for the wrappers. Otherwise, we cannot see if they suddenly break in the future.

dlaehnemann commented:
@williamrowell Thanks again for this contribution. We are currently going through all the PRs on `snakemake-wrappers` and I was wondering if you still have plans to add tests to this PR and your other open PRs? Or won't you find the time and we should see if we can amend and merge them?

williamrowell commented:
Thanks, and sorry for leaving this unfinished.  Unfortunately I won't have time for at least a few more weeks, so if you don't mind, feel free to amend and merge.  Same goes for any of my open PRs.

dlaehnemann commented:
Thanks for the update!

williamrowell commented:
I've updated this and added a test case.  It's ready for review.

dlaehnemann commented on bio/jellyfish/count/environment.yaml:
```diff
@@ -0,0 +1,6 @@
+  - conda-forge
+  - defaults
+dependencies:
+  - kmer-jellyfish==2.3.0
```
I there a particular reason for this to be pinned to `2.3.0`? Otherwise I'd opt for:
```suggestion
  - kmer-jellyfish==2.3
```

That way, bug fixes and especially stuff like security patches will automatically be used whenever available.

dlaehnemann commented on bio/jellyfish/dump/environment.yaml:
```diff
@@ -0,0 +1,6 @@
+  - conda-forge
+  - defaults
+dependencies:
+  - kmer-jellyfish==2.3.0
```
As above, I'd suggest:
```suggestion
  - kmer-jellyfish==2.3
```

dlaehnemann commented on bio/jellyfish/histo/environment.yaml:
```diff
@@ -0,0 +1,6 @@
+  - conda-forge
+  - defaults
+dependencies:
+  - kmer-jellyfish==2.3.0
```
As above, I'd suggest:
```suggestion
  - kmer-jellyfish==2.3
```

dlaehnemann commented on bio/jellyfish/merge/environment.yaml:
```diff
@@ -0,0 +1,6 @@
+  - conda-forge
+  - defaults
+dependencies:
+  - kmer-jellyfish==2.3.0
```
As above, I'd suggest:
```suggestion
  - kmer-jellyfish==2.3
```

dlaehnemann reviewed (approved):
Thanks a bunch for updating this with tests!

I added a suggestion to unpin the the patch version in the environment definition. But if you have a good reason to pin it to that exact version, that's also OK.

dlaehnemann commented:
Thanks again for revisiting this and finishing up the PR! :tada:

williamrowell commented:
Sorry for the delay.  Thanks for merging.  I updated and added tests to all but 2 of my open PRs.  The last two I need to think a little more carefully about how to construct test cases.
